### PR TITLE
User profile photos integration

### PR DESCRIPTION
**Pull request by Crxssroad**
- Created migration to add profile_photo to users table
- Added CarrierWave configurations and added public/uploads to .gitignore
- Added profile picture upload option to signup page
- Updated user model
- Added testing for profile photo uploads.

### CODE REVIEW

File: spec/features/profile_photo_spec.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    fill_in 'Last Name', with: 'Boomer'
+    fill_in "Email", with: "ash@s-mart.com"
+    fill_in "Password", with: "boomstick!3vilisd3ad"
+    fill_in "Password confirmation", with: "boomstick!3vilisd3ad"
+    attach_file :user_profile_photo, "#{Rails.root}/spec/support/images/photo.png"
+    click_button "Sign up"
+
+    expect(page).to have_content("Welcome! You have signed up successfully.")
+  end
+end
```
Comment (dovakeen118) [inline spec/features/profile_photo_spec.rb]: this feature should be added to `user_signs_up_spec`

File: db/schema.rb [hunk 2]
```diff
@@ -22,6 +22,14 @@
     t.datetime "updated_at", null: false
   end
 
+  create_table "reviews", force: :cascade do |t|
+    t.string "title", null: false
+    t.text "body", null: false
+    t.integer "rating", null: false
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+  end
```
Comment (dovakeen118) [inline db/schema.rb]: `reviews` is out of the scope of this PR for user profile photos, next time this should be part of a different PR!

File: spec/features/profile_photo_spec.rb (hunk 1 above)
Comment (dovakeen118) [inline spec/features/profile_photo_spec.rb]: we also need User model tests!

File: db/schema.rb (hunk 2 above)
Comment (dev-fildes) [reply db/schema.rb]: This was a pull from master resulting in schema update before we started.
